ironfish/src/metrics/metricsMonitor.ts

```diff
@@ -80,6 +83,15 @@ export class MetricsMonitor {
     return meter
   }
 
+  recordP2PMessage(peerId: Identity): void {
+    if (this.p2p_OutboundTrafficByPeer.has(peerId)) {
+      this.p2p_OutboundTrafficByPeer.get(peerId)?.add(1)
+    } else {
+      this.p2p_OutboundTrafficByPeer.set(peerId, this.addMeter())
+      this.recordP2PMessage(peerId)
+    }
```

Comment:
This is a bit of strange recursion. We could simplify this here

```suggestion
    const meter = this.p2p_OutboundTrafficByPeer.get(peerId)

    if(!meter) {
        meter = this.addMeter()
        this.p2p_OutboundTrafficByPeer.set(peerId, meter)
    }

    meter.add(1)
```

Reply:
Yea fair. I added the recursion when I was doing a two level deep map by peerID and message type. But with just a one level deep map it doesn't help much